source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -655,6 +665,11 @@ void Filter::onIdleTimeout() {
   read_callbacks_->connection().close(Network::ConnectionCloseType::NoFlush);
 }
 
+void Filter::onMaxConnectionDuration() {
+  ENVOY_CONN_LOG(debug, "max connection duration reached", read_callbacks_->connection());
```

Comment:
Let's also increment a stat, so folks can see how often this is getting triggered.   Possibly also worth a response flag though given we don't have one for idle timeout I guess that's optional.

Reply:
I have incremented stats. Could you please clarify how to set a response flag for that?

Reply:
still optional, but something like
getStreamInfo().setResponseFlag(StreamInfo::ResponseFlag::DurationTimeout) 
would be great.

Reply:
Done.

Reply:
sorry, response details are not the same thing as response flags, and are only intended for L7 HTTP.
you can either add a response flag for duration timeout, or just revert this line (since it was optional)

Reply:
I would like to add a response flag in https://github.com/envoyproxy/envoy/blob/83276dc7dade93a4f65c2d7de8f49637bf73e6f9/include/envoy/stream_info/stream_info.h#L36. Could you please suggest value for this?

Reply:
Done.